pymeasure/adapters/adapter.py

```diff
@@ -114,6 +119,33 @@ def binary_values(self, command, header_bytes=0, dtype=np.float32):
         raise NameError("Adapter (sub)class has not implemented the "
                         "binary_values method")
 
+    def __getattribute__(self, name):
+        """ Override to acquire the 'connection_lock' lock through the duration of attribute access.
+        """
+        if not name == "connection_lock" and \
+                object.__getattribute__(self, "connection_lock_initialized"):
+            with object.__getattribute__(self, "connection_lock"):
+                ret = object.__getattribute__(self, name)
+        else:
+            ret = object.__getattribute__(self, name)
+
+        return ret
+
+    def __setattr__(self, name, value):
+        """ Override to acquire the 'connection_lock' lock through the duration of attribute setting.
+        """
+        if not name == "connection_lock" and \
+                object.__getattribute__(self, "connection_lock_initialized"):
+            with object.__getattribute__(self, "connection_lock"):
+                object.__setattr__(self, name, value)
+
+        elif name == "connection_lock" and \
+                object.__getattribute__(self, "connection_lock_initialized"):
+            raise ValueError("Adapter connection_lock cannot be modified!")
+
+        else:
+            object.__setattr__(self, name, value)
```

Comment:
```suggestion
            super().__setattr__(name, value)
```
Why not using `super` ?

Reply:
Yep, this is definitely better. I've changed it to use `super()`